### PR TITLE
Empty address check for fail fast in client config

### LINKED ISSUE #11242: [client] networkconfig address validation missing
The following program runs fine even though no valid address has been added.

```
public class Main {

    public static void main(String[] args){
        HazelcastInstance server = Hazelcast.newHazelcastInstance();

        ClientConfig clientConfig = new ClientConfig();
        clientConfig.getNetworkConfig().addAddress("");

        HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
        System.out.println(client.getAtomicLong("foo").incrementAndGet());
    }
}
```

So we violate fail-fast.

We should check if the passed address is a valid ip address or hostname.

### PR DESCRIPTION
**Pull request by sancar**
fixes #11242

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java [hunk 1]
```diff
@@ -189,6 +190,9 @@ public ClientNetworkConfig setConnectionTimeout(int connectionTimeout) {
      * @return configured {@link com.hazelcast.client.config.ClientNetworkConfig} for chaining
      */
     public ClientNetworkConfig addAddress(String... addresses) {
+        for (String address : addresses) {
+            checkHasText(address, "member must contain text");
+        }
```
Comment (tombujok) [inline hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: Could we add a proper IP/hostname validation, maybe with URL/URI class?

Comment (sancar) [reply hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: see https://github.com/hazelcast/hazelcast/issues/11242#issuecomment-327823524
URI/URL classes are falling back to identifying as hostname if not a ipv4/ipv6. We can say  that is not an IP, but we can't say that it is not a valid hostname. And that is not enough. 
And also we don't want to get into hostname resolution also to validate hostname is valid. 
If is there a way that I don't know, I am open to suggestions.

Comment (pveentjer) [reply hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: the checkHasText doesn't trim. So if somebody feeds spaces, tabs or linefeeds, these are not detected.

Comment (sancar) [reply hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: I added the trim.

File: hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java [hunk 2]
```diff
@@ -189,6 +190,9 @@ public ClientNetworkConfig setConnectionTimeout(int connectionTimeout) {
      * @return configured {@link com.hazelcast.client.config.ClientNetworkConfig} for chaining
      */
     public ClientNetworkConfig addAddress(String... addresses) {
+        for (String address : addresses) {
+            checkHasText(address.trim(), "member must contain text");
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: In theory someone could pass a null as address and you will be causing an 'uncontrolled' NPE.

Comment (Donnerbart) [reply hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: Programmatically this should be very easy to accomplish, shouldn't it? Would be good to have a distinct test case for this.

Comment (sancar) [reply hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: fixed and added the test.

Comment (Donnerbart) [reply hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: I think this won't help, due to the `(String... addresses)` parameter, so `config.getNetworkConfig().addAddress(null, null);` should still through an `NPE`. I think the null check has to be done in the loop.

**Comment by sancar** [comment]
run-lab-run

File: hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java (hunk 2 above)
Comment (sancar) [reply hazelcast-client/src/main/java/com/hazelcast/client/config/ClientNetworkConfig.java]: @Donnerbart fixed again :(

**Comment by pveentjer** [comment]
In the future we could perhaps add a stronger check. But I'm already happy with the current validation.

**Pull request merged**